ignite/distributed/utils.py

```diff
@@ -350,6 +353,36 @@ def all_reduce(
     return _model.all_reduce(tensor, op, group=group)
 
 
+def all_gather_tensors_with_shapes(
+    tensor: torch.Tensor, shapes: Sequence[Sequence[int]], group: Optional[Union[Any, List[int]]] = None
+) -> List[torch.Tensor]:
+    """Gather tensors with different shapes but with the same number of dimensions from across processes."""
```

Comment:
@sadra-barikbin is it possible to provide a quick example of usage of the method and Args section?